MyLibrary/Library/Controllers/AuthorController.cs

```diff
@@ -0,0 +1,154 @@
+﻿// <copyright file="AuthorController.cs" company="Peretiatko Anastasiia">
+// Copyright (c) Peretiatko Anastasiia. All rights reserved.
+// </copyright>
+
+using System;
+using BusinessLogic.LibraryModel;
+using BusinessLogic.LibraryService;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Library.Controllers
+{
+    /// <summary>
+    /// Controller for dealing with Author
+    /// </summary>
+    /// <seealso cref="Microsoft.AspNetCore.Mvc.ControllerBase" />
+    [Route("api/[controller]")]
+    [ApiController]
+    public class AuthorController : ControllerBase
+    {
+        /// <summary>
+        /// The library service
+        /// </summary>
+        private ILibraryService libraryService;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AuthorController"/> class
+        /// </summary>
+        /// <param name="libraryService">The library services</param>
+        public AuthorController(ILibraryService libraryService)
+        {
+            this.libraryService = libraryService;
+        }
+
+        // GET: api/author
+        /// <summary>
+        /// Gets all authors from the Library
+        /// </summary>
+        /// <returns>
+        /// HTTP status code Ok() with the list of authors
+        /// or HTTP status code NoContent() if there are any authors in library
+        /// </returns>
+        [HttpGet]
+        public IActionResult GetAllAuthors()
+        {
+            if (this.libraryService.GetAllAuthors() != null)
+            {
+                return Ok(this.libraryService.GetAllAuthors());
+            }
+
+            return NoContent();
+        }
+
+        // GET: api/author/5
+        /// <summary>
+        /// Gets the author by its id
+        /// </summary>
+        /// <param name="id">The identifier of the author</param>
+        /// <returns>
+        /// HTTP status code Ok() with found author
+        /// or HTTP status code NotFound() if there isn`t author with such id
+        /// </returns>
+        [HttpGet("{id}")]
+        public IActionResult GetAuthorById(uint id)
+        {
+            Author foundAuthor = this.libraryService.GetAuthorById(id);
+            if (foundAuthor == null)
+            {
+                return NotFound("No authors with such id!");
+            }
+
+            return Ok(foundAuthor);
+        }
+
+        // POST: api/author
+        /// <summary>
+        /// Adds the new author to the author list
+        /// </summary>
+        /// <param name="author">The new author</param>
+        /// <returns>
+        /// HTTP status code Created() with added author`s id
+        /// or HTTP status code BadRequest() with error message
+        /// </returns>
+        [HttpPost]
+        public IActionResult AddNewAuthor([FromBody]Author author)
+        {
+            ///???
```

Comment:
You haven`t fixed your issue with validation?

Reply:
Validation works, but, as I understand from my tests, this part of code will never be executed.
The constructor of Author will be called when converting JSon data to Author author. Then our Data Validation Attribute will work and we will get BadRequest anyway, even without this "if", and see message from Data Validation Attribute.

Reply:
hm it strange, maybe some configuration for you project set data annotation to act in this way.  Commonly you need this if.